### PR TITLE
Content update for NetEntities

### PR DESCRIPTION
**Pull request by metalgearsloth**
Requires https://github.com/space-wizards/RobustToolbox/pull/4252

Adding helper methods to EntityManager to make conversion easy too.

### CODE REVIEW

File: Content.Client/Vehicle/VehicleSystem.cs [hunk 1]
```diff
@@ -36,10 +36,12 @@ private void OnRiderHandleState(EntityUid uid, RiderComponent component, ref Com
         if (args.Current is not RiderComponentState state)
             return;
 
+        var entity = GetEntity(state.Entity);
```
Comment (ElectroJr) [inline Content.Client/Vehicle/VehicleSystem.cs:39]: Needs to handle unknown net ids

**Review by ElectroJr — CHANGES_REQUESTED** [review]
Only a partial review (100/486). I left comments on any comp states that should get updated to somehow handle unknown entities. Some of those comp states might work fine as is, but its probably safer to still update them and set good examples. E.g., the rider comp state implies that the rider is a child of the vehicle, so the vehicle should never be an unknown entity.

File: Content.Server/Anomaly/AnomalySystem.Commands.cs [hunk 2]
```diff
@@ -26,7 +26,7 @@ private void PulseAnomalyCommand(IConsoleShell shell, string argstr, string[] ar
         if (args.Length != 1)
             shell.WriteError("Argument length must be 1");
 
-        if (!EntityUid.TryParse(args[0], out var uid))
+        if (!NetEntity.TryParse(args[0], out var uidNet) || !TryGetEntity(uidNet, out var uid))
```
Comment (ElectroJr) [inline Content.Server/Anomaly/AnomalySystem.Commands.cs:29]: Possibly worth adding a helper method for this?

File: Content.Server/Atmos/EntitySystems/GasTileOverlaySystem.cs [hunk 3]
```diff
@@ -330,17 +331,19 @@ private void UpdatePlayer(IPlayerSession playerSession, GameTick curTick)
...
             {
+                var grid = GetEntity(netGrid);
+
                 // Not all grids have atmospheres.
                 if (!TryComp(grid, out GasTileOverlayComponent? overlay))
                     continue;
 
                 List<GasOverlayChunk> dataToSend = new();
-                ev.UpdatedChunks[grid] = dataToSend;
+                ev.UpdatedChunks[GetNetEntity(grid)] = dataToSend;
```
Comment (ElectroJr) [inline Content.Server/Atmos/EntitySystems/GasTileOverlaySystem.cs:346]: you already have netGrid here

File: Content.Shared/DoAfter/SharedDoAfterSystem.cs [hunk 4]
```diff
@@ -195,6 +206,21 @@ public bool TryStartDoAfter(DoAfterArgs args, [NotNullWhen(true)] out DoAfterId?
...
+        // Networking yay
+        args.NetTarget = GetNetEntity(args.Target);
+        args.NetUsed = GetNetEntity(args.Used);
+        args.NetUser = GetNetEntity(args.User);
+        args.NetEventTarget = GetNetEntity(args.EventTarget);
+
+        if (!args.User.IsValid())
+        {
+            // Weh
+        }
```
Comment (ElectroJr) [inline Content.Shared/DoAfter/SharedDoAfterSystem.cs:222]: Weh

Comment (metalgearsloth) [reply Content.Shared/DoAfter/SharedDoAfterSystem.cs:222]: debugging be like

**Review by ElectroJr — APPROVED** [review]

**Pull request merged**